examples/remote/binary_dependencies/cargo/crates.bzl

```diff
@@ -9,6 +9,144 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")  # bui
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")  # buildifier: disable=load
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")  # buildifier: disable=load
 
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of normal dependencies for the Rust targets of that package.
+_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+        "ferris-says": "@remote_binary_dependencies__ferris_says__0_2_0//:ferris_says",
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of proc_macro dependencies for the Rust targets of that package.
+_PROC_MACRO_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of normal dev dependencies for the Rust targets of that package.
+_DEV_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of proc_macro dev dependencies for the Rust targets of that package.
+_DEV_PROC_MACRO_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+def crates(deps):
+    """EXPERIMENTAL -- MAY CHANGE AT ANY TIME: Finds the fully qualified label of the requested crates for the package where this macro is called.
+
+    WARNING: This macro is part of an expeirmental API and is subject to change.
+
+    Args:
+        deps (list or str): Either a list of dependencies or a string of one which will
```

Comment:
Can this be more specific? Is this a crate name? Does it contain a version?

Reply:
It's a crate name. Are you asking why `deps` is either a list or a string? I thought it would be more convenient as I know I and others have run into situations where something expects a list but a string was given instead. 

The equivalent thing for this field is something like:
```python
load("//third_party/cargo:crates.bzl", "crates")
rust_library(
    name = "example",
    deps = [
        "//third_party/cargo:dependency1",
        crates("dependency2"),
    ],
    #...
)
```

The two dependencies are pulled from the same `cargo-raze` output, just using different methods of doing so.

Reply:
No. I mean, as a reader, "list of dependencies" is insufficiently specific. Is it a crate identifier?

Reply:
I wrote some pseudo code below where I called this argument "crate_names". How do you feel about that option.

Reply:
"crate_names" is Ok. The names I chose were inspired by `rules_python'`s `requirements()` and `all_requirements`. I wanted `crates` and `all_crates()` to represent the same thing. Granted there is a bit more "magic" here that in `rules_python` but I don't think it's that much more (especially now that `package_name` is an argument). I generally prefer verbosity as well but I thought that the current names matched an existing and common pattern and would be more intuitive. What are your thoughts on the difference between this and what `rules_python` is doing?

Reply:
I better understand where this design came from having familiarized myself with rules_python.

I still prefer crate_names though. "deps" could either be expected to be a build target, a bare crate name, or a crate-crate_version.